bumble/profiles/asha_service.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2021-2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# -----------------------------------------------------------------------------
+# Imports
+# -----------------------------------------------------------------------------
+import struct
+from ..device import (
+  Device, AdvertisingType
+)
+from ..core import AdvertisingData
+from ..gatt import (
+  GATT_ASHA_SERVICE,
+  GATT_ASHA_READ_ONLY_PROPERTIES_CHARACTERISTIC,
+  GATT_ASHA_AUDIO_CONTROL_POINT_CHARACTERISTIC,
+  GATT_ASHA_AUDIO_STATUS_CHARACTERISTIC,
+  GATT_ASHA_VOLUME_CHARACTERISTIC,
+  GATT_ASHA_LE_PSM_OUT_CHARACTERISTIC,
+  TemplateService,
+  Characteristic,
+  CharacteristicValue,
+  PackedCharacteristicAdapter
+)
+
+
+# -----------------------------------------------------------------------------
+class AshaService(TemplateService):
+  UUID = GATT_ASHA_SERVICE
+
+  def __init__(self,device:Device):
+    self.device=device
+
+    # Handler for volume control
+    def on_volume_write(connection, value):
+      print('--- VOLUME Write:', value[0])
```

Comment:
use `logger.debug()` instead of directly `print()` (same for other uses of `print()` in this file). See other files in the project for examples of how to instantiate a `logger` object with the right namespace.